pkg/core/resources/apis/mesh/dataplane_helpers.go

```diff
@@ -260,3 +260,11 @@ func (d *DataplaneResource) AsOutbounds(resolver core_model.LabelResourceIdentif
 	}
 	return outbounds
 }
+
+func (d *DataplaneResource) GetProxyType() mesh_proto.ProxyType {
+	spec := d.GetSpec().(*mesh_proto.Dataplane)
+	if spec.IsDelegatedGateway() || spec.IsBuiltinGateway() {
```

Comment:
we agreed in the madr that delegated is going to have type `Sidecar`, that's how `targetRef.proxyType` field works today

Reply:
changed